api/v1alpha1/imageclusterinstall_webhook.go

```diff
@@ -147,3 +150,10 @@ func isValidMachineNetwork(machineNetwork string) error {
 	}
 	return nil
 }
+
+func isValidNoProxy(proxy *Proxy, machineNetwork string) error {
```

Comment:
I think there is small misunderstanding, it should be relevant to any proxy field and not for no proxy specifically.
More of it , no proxy field is the less relevant here, we should check if http or https proxy was set

Reply:
[This](https://issues.redhat.com/browse/MGMT-19107?focusedId=25926359&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25926359) comment made by Nick, made me think the issue was only with no_proxy. If that's not the case I'll change it.

Reply:
The idea was that we need machine network for creating right no proxy but no proxy is do nothing by itself.
We need to verify http and https proxy params as they actually set proxy

Reply:
Nope. The real issue is that if http or https proxy is defined we require machine network